R/dash.R

```diff
@@ -1439,8 +1461,14 @@ Dash <- R6::R6Class(
       # insert meta tags if present
       meta_tags <- all_tags[["meta_tags"]]
 
-      private$.index <- sprintf(
-        '<!DOCTYPE html>
+      if (!is.null(private$custom_index)) {
+        interpolated_index <- glue::glue(private$custom_index)
```

Comment:
@HammadTheOne To be honest, I'm not quite sure what this call to `glue::glue` is doing. 

If it's simple concatenation, I'd replace with a call to `paste0` or `sprintf` to avoid pulling in the dependency. Something like

```
> test <- c("foo", "bar")
paste0("A few things to worry about: ", paste0(test, collapse=", "))
[1] "A few things to worry about: foo, bar"
```

or

```
sprintf("A few things to worry about: %s", paste(test, collapse = ", "))
```

...though generally we try to avoid mixing `sprintf` and `paste`.

Reply:
In this case, I was thinking of using `glue` since it can handle string interpolation for us, and is syntactically similar to the way `index_string` is handled in the Python implementation using braces for the required keys and adding in the `meta_tags`, `scripts_tags` variables that are already declared for the default index. 

Based on that, is it still something that would be better changed to base R functions or should it remain?

Reply:
@HammadTheOne I think this is OK here, especially given that we're already pulling in this package because of another Dash dependency.

Reply:
@HammadTheOne Actually, upon reflection, I don't think we should use `glue::glue` as-is, because of something noted here previously:

https://github.com/plotly/dash/pull/286#discussion_r201783101

We want the keys to be enclosed with `%`, and I think `glue` can't handle that, since it parses their contents as executable R code. Typically `%` in R is used to enclose operators to be applied to matrices, e.g. `%*%` for matrix multiplication. So `glue` will fail:

```
> mykey <- "test"
> glue::glue("this is a {mykey}")
this is a test
> glue::glue("this is a {%mykey%}")
Error in parse(text = text, keep.source = FALSE) : 
  <text>:1:1: unexpected SPECIAL
1: %mykey%
    ^
```

Reply:
I think this is a great change for parity, that way people can even use the same index string between Python and R apps if they want. Added/fixed in 4023b6.